src/contrib/cluster/Akka.Cluster.Tools/Singleton/MemberAgeOrdering.cs

```diff
@@ -15,15 +15,25 @@ namespace Akka.Cluster.Tools.Singleton
     internal sealed class MemberAgeOrdering : IComparer<Member>
     {
         private readonly bool _ascending;
+        private readonly bool _considerAppVersion;
 
-        private MemberAgeOrdering(bool ascending)
+        private MemberAgeOrdering(bool ascending, bool considerAppVersion)
         {
             _ascending = ascending;
+            _considerAppVersion = considerAppVersion;
         }
 
         /// <inheritdoc/>
         public int Compare(Member x, Member y)
         {
+            if (_considerAppVersion)
+            {
+                // prefer nodes with the highest app version, even if they're younger
+                var appVersionDiff = x.AppVersion.CompareTo(y.AppVersion);
+                if (appVersionDiff != 0)
```

Comment:
So we have to run the sort in two phases here:

1. AppVersion
2. Age within the AppVersion cohort.

Looks to me like the sorter disregards age when `AppVersion` is considered - is that correct?

Reply:
This design is wrong then - without the member age requirement there's no guarantee of stable singleton placement, which defeats the purpose of adding this feature (i.e. being able to predict the next location of where a singleton will be placed during a deployment)

We need to do the sort in multiple phases:

1. Members by role
2. Members within a role that have the highest app version
3. Oldest member within the results of phase 2.

Let's design it to work that way.